src/lib/intent/base-intent.ts

```diff
@@ -115,6 +121,20 @@ export default abstract class Intent implements IntentInterface {
     return true
   }
 
+  /**
+   * @returns The API action to perform on save. Defaults to saving the Content Type.
+   */
+  getSaveAction (): APIAction {
+    return new ContentTypeSaveAction(this.getContentTypeId())
+  }
+
+  /**
+   * @returns The API action to perform on publish. Defaults to publishing the Content Type.
+   */
+  getPublishAction (): APIAction {
```

Comment:
This block doesn't seem to be used at all?
Also you can just import the `ContentTypeSaveAction` / `ContentTypePublishAction` in your intent and add it as an action. See `src/lib/intent/field-delete.ts` for an example.